File: 02_weather_app/.claude/skills/format-json/scripts/format_json.js
#!/usr/bin/env node
/**
 * Pretty-print a JSON file in place with 2-space indentation.
 * Usage: node format_json.js <path-to-json-file>
 */
const fs = require("fs");

const path = process.argv[2];
if (!path) {
  console.error("Usage: node format_json.js <path-to-json-file>");
  process.exit(1);
}

const raw = fs.readFileSync(path, "utf8");

let data;
try {
  data = JSON.parse(raw);
} catch (e) {
  console.error(`Invalid JSON in ${path}: ${e.message}`);
  process.exit(1);
}

fs.writeFileSync(path, JSON.stringify(data, null, 2) + "\n");
console.log(`Formatted ${path}`);

File: 02_weather_app/.claude/skills/format-json/scripts/format_json.py
#!/usr/bin/env python3
"""Pretty-print a JSON file in place with 2-space indentation.

Usage: python format_json.py <path-to-json-file>
"""
import json
import sys


def main():
    if len(sys.argv) != 2:
        print("Usage: python format_json.py <path-to-json-file>", file=sys.stderr)
        sys.exit(1)

    path = sys.argv[1]

    with open(path, "r", encoding="utf-8") as f:
        try:
            data = json.load(f)
        except json.JSONDecodeError as e:
            print(f"Invalid JSON in {path}: {e}", file=sys.stderr)
            sys.exit(1)

    with open(path, "w", encoding="utf-8") as f:
        f.write(json.dumps(data, indent=2))
        f.write("\n")

    print(f"Formatted {path}")


if __name__ == "__main__":
    main()
